### ISSUE TITLE
Must the last layer of mxnet model be output layers like Softmax, LinearRegression, etc?

### ISSUE THREAD

**Issue by yxchng** [apache/incubator-mxnet#3460]
Must the last layer of mxnet model be output layers like Softmax, LinearRegression, etc? I want to create a model where the output layer is actually a group of the outputs of convolution layers, however I bump into the problem of my data label not matching:

 Symbol.InferShapeKeyword argument name heatmap not found.

After my research on the issues that other posted, it seems that the name of the output layer should match the name of my output layer. However, Group layer does not have a name. My group layer consists of [c1, c2,..,c8] where each of the c's are output of a convolution layer, hence they are images.

All the examples on mxnet seems to have output layers defined in mxnet like Softmax and LinearRegression as the last layer.

Is Group as the last layer not supported?

For reference, the following is my code:

``` python
def createModel(nStack, nModules):
    inp = mx.symbol.Variable('data')

    cnv1_ = mx.symbol.Convolution(inp, num_filter = 64, kernel = (7, 7), stride = (2, 2), pad = (3, 3))
    cnv1 = mx.symbol.Activation(mx.symbol.BatchNorm(data = cnv1_), act_type = 'relu')
    r1 = residual(cnv1, 128, dim_match = False)
    pool = mx.symbol.Pooling(r1, kernel = (2, 2), stride = (2, 2), pool_type = 'max')
    r4 = residual(pool, 128, dim_match = True)
    r5 = residual(r4, 256, dim_match = False)

    inter = r5
    out = []

    for i in xrange(nStack):
        hg = hourglass(inter, 4, 256, nModules)
        for i in xrange(nModules):
            ll = residual(hg, 256, dim_match = True)
        ll = _conv_bn_relu(ll, num_filter = 256)
        tmpOut = mx.symbol.Convolution(ll, num_filter = 16, kernel = (1, 1), stride = (1, 1))
        out.append(tmpOut)

        if i < nStack:
            ll_ = mx.symbol.Convolution(ll, num_filter = 256, kernel = (1, 1), stride = (1, 1))
            tmpOut_ = mx.symbol.Convolution(tmpOut, num_filter = 256, kernel = (1, 1), stride = (1, 1))
            inter = inter + ll_ + tmpOut_

    group = mx.symbol.Group(out)

    return group

```

**Comment by piiswrong** [comment]
output doesn't have to match anything. However if it's not loss layer you need to provide head gradient for backprop
if your input has a symbol called heatmap your data iterator must output that

**Comment by yxchng** [comment]
Can you reference me to an example where head gradient for backprop is provided? I can't seem to find an example.

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
